src/app/collective-rewards/rewards/utils/formatMetrics.test.ts

```diff
@@ -1,64 +1,55 @@
-  })
+describe('formatMetrics', () => {
+  it('should return `fiatAmount` equals to 0 if `amount` is zero', () => {
+    const result = formatMetrics(0, 10, 'RIF', 'USD')
 
-  it('should format onchain fraction correctly with custom decimals', () => {
-    const amount = BigInt('123456789')
-    const result = formatOnchainFraction(amount, 2, 9)
-    expect(result).toBe('0.12')
+    expect(result.fiatAmount).toBe(formatFiatAmount(0, 'USD'))
   })
+  it('should return `fiatAmount` equals to 0 if `price` is zero', () => {
+    const result = formatMetrics(10, 0, 'RIF', 'USD')
 
-  it('should handle zero amount correctly', () => {
-    const amount = BigInt('0')
-    const result = formatOnchainFraction(amount)
-    expect(result).toBe('0')
+    expect(result.fiatAmount).toBe(formatFiatAmount(0, 'USD'))
   })
+  it('should return `amount` equals to 0 if `amount` is zero ', () => {
+    const result = formatMetrics(0, 10, 'RIF', 'USD')
 
-  it('should handle numbers without decimals', () => {
-    const amount = BigInt('9000000000000000000')
-    const result = formatOnchainFraction(amount)
-    expect(result).toBe('9')
+    expect(result.amount).toBe(formatAmount(0, 'RIF'))
   })
+  it('should accept value as different data types(amount)', () => {
+    const valueAsNumber = 1.23456789
+    const valueAsBigInt = 1234567890000000000n
 
-  it('should handle numbers without decimals and more than 1 character', () => {
-    const amount = BigInt('90000000000000000000')
-    const result = formatOnchainFraction(amount)
-    expect(result).toBe('90')
-  })
+    const resultBigInt = formatMetrics(valueAsBigInt, 10, 'RIF', 'USD')
+    const resultNumber = formatMetrics(valueAsNumber, 10, 'RIF', 'USD')
 
-  it('should handle numbers with decimals and 3 characters', () => {
-    const amount = BigInt('900561000000000000000')
-    const result = formatOnchainFraction(amount)
-    expect(result).toBe('900.56')
+    expect(resultBigInt.amount).toBe(formatAmount(1.23, 'RIF'))
+    expect(resultBigInt.fiatAmount).toBe(formatFiatAmount(12.346, 'USD'))
+    expect(resultNumber.amount).toBe(formatAmount(1.23, 'RIF'))
+    expect(resultNumber.fiatAmount).toBe(formatFiatAmount(12.346, 'USD'))
   })
+  it('should accept value as different data types(price)', () => {
+    const resultBigInt = formatMetrics(1.23456789, 10n, 'RIF', 'USD')
+    const resultNumber = formatMetrics(1.23456789, 10, 'RIF', 'USD')
 
-  it('should handle numbers with decimals and 3 characters', () => {
-    const amount = BigInt('9000127000000000000000')
-    const result = formatOnchainFraction(amount)
-    expect(result).toBe('9000.12')
+    expect(resultBigInt.amount).toBe(formatAmount(1.23, 'RIF'))
+    expect(resultBigInt.fiatAmount).toBe(formatFiatAmount(12.346, 'USD'))
+    expect(resultNumber.amount).toBe(formatAmount(1.23, 'RIF'))
+    expect(resultNumber.fiatAmount).toBe(formatFiatAmount(12.346, 'USD'))
   })
+  it.only('should format RIF properly', () => {
```

Comment:
I start introduction the tests, but I found out the following challenge that we may need to change. 

If I do `formatCurrency(10,USD)` it returns 0. 

 If I want to represent a 10 I need to do `formatCurrency(10.000000000000000000,USD)` or `formatCurrency(10000000000000000000,USD)`.

Im not sure if this is the way that it should work but its confusing.

Reply:
@antomor